rust/worker/src/storage/s3.rs

```diff
@@ -313,26 +395,98 @@ mod tests {
+        let client = get_s3_client();
 
         let storage = S3Storage {
             bucket: "test".to_string(),
             client,
         };
         storage.create_bucket().await.unwrap();
 
-        // Write some data to a test file, put it in s3, get it back and verify its contents
-        let tmp_dir = tempdir().unwrap();
-        let persist_path = tmp_dir.path().to_str().unwrap().to_string();
-
         let test_data = "test data";
-        let test_file_in = format!("{}/test_file_in", persist_path);
-        std::fs::write(&test_file_in, test_data).unwrap();
-        storage.put_file("test", &test_file_in).await.unwrap();
-        let mut bytes = storage.get("test").await.unwrap();
+        storage
+            .put_bytes("test", test_data.as_bytes().to_vec())
+            .await
+            .unwrap();
 
-        let mut buf = String::new();
-        bytes.read_to_string(&mut buf).await.unwrap();
+        let mut stream = storage.get("test").await.unwrap();
+
+        let mut buf = Vec::new();
+        while let Some(chunk) = stream.next().await {
+            match chunk {
+                Ok(data) => {
+                    buf.extend_from_slice(&data);
+                }
+                Err(e) => {
+                    panic!("Error reading stream: {}", e);
+                }
+            }
+        }
+
+        let buf = String::from_utf8(buf).unwrap();
         assert_eq!(buf, test_data);
     }
+
+    async fn test_put_file(file_size: usize) {
+        let client = get_s3_client();
+
+        let storage = S3Storage {
+            bucket: "test".to_string(),
+            client,
+        };
+        storage.create_bucket().await.unwrap();
+
+        let mut temp_file = NamedTempFile::new().unwrap();
+
+        let mut rng = rand_xorshift::XorShiftRng::seed_from_u64(0);
+        let mut remaining_file_size = file_size;
+
+        while remaining_file_size > 0 {
+            let chunk_size = std::cmp::min(remaining_file_size, 4096);
+            let mut chunk = vec![0u8; chunk_size];
+            rng.try_fill(&mut chunk[..]).unwrap();
+            temp_file.write_all(&chunk).unwrap();
+            remaining_file_size -= chunk_size;
+        }
+
+        storage
+            .put_file("test", &temp_file.path().to_str().unwrap())
+            .await
+            .unwrap();
+
+        let mut stream = storage.get("test").await.unwrap();
+
+        let mut buf = Vec::new();
+        while let Some(chunk) = stream.next().await {
+            match chunk {
+                Ok(data) => {
+                    buf.extend_from_slice(&data);
+                }
+                Err(e) => {
+                    panic!("Error reading stream: {}", e);
+                }
+            }
+        }
+
+        let file_contents = std::fs::read(temp_file.path()).unwrap();
+        assert_eq!(buf, file_contents);
+    }
+
+    #[tokio::test]
+    #[cfg(CHROMA_KUBERNETES_INTEGRATION)]
+    async fn test_put_file_scenarios() {
```

Comment:
I wanted to use proptest but apparently proptest doesn't work for async functions yet :/ 

I guess I could `block_on` if that's preferred

Reply:
I think some unit tests are good enough here